File: backend/app/Services/TypesInvites.php
<?php

namespace App\Services;

/**
 * Types d'invites d'une reservation.
 *
 * La base stocke des codes anglais - c'est un enum, on n'y touche pas.
 * Ce qui s'affiche, en revanche, doit parler a l'agent comme au client :
 * le francais, puis l'arabe entre parentheses.
 */
class TypesInvites
{
    private const LIBELLES = [
        "Males"                  => ["Hommes", "\u{631}\u{62C}\u{627}\u{644}"],
        "Females"                => ["Femmes", "\u{646}\u{633}\u{627}\u{621}"],
        "Family"                 => ["Famille", "\u{639}\u{627}\u{626}\u{644}\u{629}"],
        "Professional visitors"  => ["Visiteurs professionnels", "\u{632}\u{648}\u{627}\u{631} \u{645}\u{647}\u{646}\u{64A}\u{648}\u{646}"],
    ];

    /** Les codes tels que la base les attend. */
    public static function codes(): array
    {
        return array_keys(self::LIBELLES);
    }

    /** "Hommes (رجال)" ; le code inconnu se rend tel quel. */
    public static function libelle(?string $code): string
    {
        if ($code === null || $code === "") {
            return "";
        }
        if (!isset(self::LIBELLES[$code])) {
            return $code;
        }
        [$fr, $ar] = self::LIBELLES[$code];
        return $fr . " (" . $ar . ")";
    }

    /** Le francais seul, pour les colonnes etroites. */
    public static function francais(?string $code): string
    {
        if ($code === null || $code === "" || !isset(self::LIBELLES[$code])) {
            return $code ?? "";
        }
        return self::LIBELLES[$code][0];
    }

    /** Liste destinee aux applications : code, francais, arabe, libelle. */
    public static function liste(): array
    {
        $sortie = [];
        foreach (self::LIBELLES as $code => [$fr, $ar]) {
            $sortie[] = [
                "code"     => $code,
                "francais" => $fr,
                "arabe"    => $ar,
                "libelle"  => $fr . " (" . $ar . ")",
            ];
        }
        return $sortie;
    }
}

File: mobile/immobilier-sakane/lib/core/constants/types_invites.dart
/// Types d'invites d'une reservation.
///
/// La base attend des codes anglais - c'est un enum SQL, on ne le
/// touche pas. Ce que l'agent lit, en revanche, doit lui parler : le
/// francais, puis l'arabe entre parentheses.
class TypeInvite {
  final String code;
  final String francais;
  final String arabe;

  const TypeInvite(this.code, this.francais, this.arabe);

  /// « Hommes (\u0631\u062C\u0627\u0644) »
  String get libelle => "$francais ($arabe)";

  static const List<TypeInvite> tous = [
    TypeInvite("Males", "Hommes", "\u0631\u062C\u0627\u0644"),
    TypeInvite("Females", "Femmes", "\u0646\u0633\u0627\u0621"),
    TypeInvite("Family", "Famille", "\u0639\u0627\u0626\u0644\u0629"),
    TypeInvite("Professional visitors", "Visiteurs professionnels",
        "\u0632\u0648\u0627\u0631 \u0645\u0647\u0646\u064A\u0648\u0646"),
  ];

  /// Les codes, dans l'ordre d'affichage.
  static List<String> get codes => tous.map((t) => t.code).toList();

  /// Le libelle d'un code ; un code inconnu se rend tel quel plutot
  /// que de disparaitre.
  static String libelleDe(String? code) {
    if (code == null || code.isEmpty) return "";
    for (final t in tous) {
      if (t.code == code) return t.libelle;
    }
    return code;
  }
}
